transformer_engine/common/fused_attn/fused_attn_max_512.cu

```diff
@@ -0,0 +1,1555 @@
+/*************************************************************************
+ * Copyright (c) 2022-2023, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+ *
+ * See LICENSE for license information.
+ ************************************************************************/
+
+#include "fused_attn_max_512.h"
+
+#include <cuda_bf16.h>
+#include <cuda_fp16.h>
+#include <map>
+#include <unordered_map>
+#include <vector>
+
+#include "cudnn_frontend.h"
+
+#include "../common.h"
+#include "utils.h"
+
+#define CUDNN_FRONTEND_UNUSED(X) ((void)X)
+
+#define Q_ID 1
+#define K_ID 2
+#define V_ID 3
+#define O_ID 4
+#define S_ID 5
+#define B_ID 6
+#define D_CONST_ID 7
+#define S_CONST_ID 8
+#define Q_SEQLEN_ID 9
+#define K_SEQLEN_ID 10
+#define dQ_ID 11
+#define dK_ID 12
+#define dV_ID 13
+#define dO_ID 14
+#define MASK_VAL_ID 15
+#define dS_ID 16
+#define dBias_ID 17
+
+#define VIRTUAL_ID 20
+
+namespace transformer_engine {
+namespace fused_attn {
+
+#if (CUDNN_VERSION >= 8700)
```

Comment:
Should this be 8901 to be consistent with the fwd/bwd kernels?

Reply:
Fixed in https://github.com/NVIDIA/TransformerEngine/pull/175/commits/857ef2e232534a96c9e87d9fe25aab1a0d282f54